portfolio/src/main/webapp/script.js

```diff
@@ -40,21 +39,20 @@ async function onLoad() {
 
 async function login() {
     const response = await fetch('/login');
-    const welcome = await response.json();
-    if (welcome[0] === "") {
-        loginUrl = welcome[1];
-        document.getElementById('login-container').innerHTML = " <a href="+ welcome[1] +"> Login </a>";
+    const responseData = await response.json();
+    if (responseData["email"] === "") {
```

Comment:
If you are checking whether there is an "email" field in `responseData`, then you'd better use `if (responseData["email"])` instead. As if there is no that field, the return value would be `undefined` instead of empty string.